system_tests/stylus_tracer_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2024, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE
+
+package arbtest
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/google/go-cmp/cmp"
+	"github.com/offchainlabs/nitro/execution/gethexec"
+	"github.com/offchainlabs/nitro/solgen/go/mocksgen"
+	"github.com/offchainlabs/nitro/util/testhelpers"
+)
+
+func TestStylusTracer(t *testing.T) {
+	const jit = false
+	builder, auth, cleanup := setupProgramTest(t, jit)
+	ctx := builder.ctx
+	l2client := builder.L2.Client
+	l2info := builder.L2Info
+	rpcClient := builder.L2.Client.Client()
+	defer cleanup()
+
+	traceTransaction := func(tx common.Hash, tracer string) []gethexec.HostioTraceInfo {
+		traceOpts := struct {
+			Tracer string `json:"tracer"`
+		}{
+			Tracer: tracer,
+		}
+		var result []gethexec.HostioTraceInfo
+		err := rpcClient.CallContext(ctx, &result, "debug_traceTransaction", tx, traceOpts)
+		Require(t, err, "trace transaction")
+		return result
+	}
+
+	// Deploy contracts
+	stylusMulticall := deployWasm(t, ctx, auth, l2client, rustFile("multicall"))
+	evmMulticall, tx, _, err := mocksgen.DeployMultiCallTest(&auth, builder.L2.Client)
+	Require(t, err, "deploy evm multicall")
+	_, err = EnsureTxSucceeded(ctx, l2client, tx)
+	Require(t, err, "ensure evm multicall deployment")
+
+	// Args for tests
+	key := testhelpers.RandomHash()
+	value := testhelpers.RandomHash()
+	loadStoreArgs := multicallEmptyArgs()
+	loadStoreArgs = multicallAppendStore(loadStoreArgs, key, value, false)
+	loadStoreArgs = multicallAppendLoad(loadStoreArgs, key, false)
+	callArgs := argsForMulticall(vm.CALL, stylusMulticall, nil, []byte{0})
+	evmCall := argsForMulticall(vm.CALL, evmMulticall, nil, []byte{0})
+
+	for _, testCase := range []struct {
```

Comment:
nice!

Reply:
Not for now - but after both this and stylus_EVM are merged, we might be able to re-use some testcases between the two